File: gccws-server/comon/src/main/java/com/gccws/common/service/MessageHistoryService.java
package com.gccws.common.service;
import com.gccws.common.model.CommonDropdownModel;
import com.gccws.common.model.CommonPageableData;
import com.gccws.base.service.BaseService;
import com.gccws.common.dto.MessageHistoryDto;

import java.util.List;

/**
 * @author  Md. Mizanur Rahman
 * @Email   kmmizanurrahmanjp@gmail.com
 * @since   March 14,2023
 * @version 1.0
 */

public interface MessageHistoryService extends BaseService<MessageHistoryDto, CommonDropdownModel, CommonPageableData> {

    List<MessageHistoryDto> getByTransactionIdAndTransactionTable(Integer transactionId, String transactionTable);
    List<MessageHistoryDto> getByUserId(Integer userId);

}

File: gccws-ui/src/app/main/shared/service/message-history.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {AppCrudRequestService} from "../../core/service/app-crud-request.service";
import {AppUtils} from "../../core/utils/app.utils";
import {AppConstants} from "../../core/constants/app.constants";
import {MessageHistory} from "../model/message-history";
import {Observable} from "rxjs";
import {CommonResponseList} from "../../core/model/common-response";

/**
 * @author	Md. Mizanur Rahman
 * @Email	  kmmizanurrahmanjp@gmail.com
 * @since	  March 14,2023
 * @version 1.0
 */

@Injectable({
  providedIn: 'root'
})
export class MessageHistoryService extends AppCrudRequestService<MessageHistory>{

  constructor(
    private http: HttpClient,
    private appUtils: AppUtils,
    private appConstants: AppConstants,
  ) {
    super(http, appUtils.getPrivateBaseURL() + appConstants.moduleCommon + 'message-history');
  }

  getByUserId(): Observable<CommonResponseList<MessageHistory>> {
    return this.http.get<CommonResponseList<MessageHistory>>( this._BASE_URL + '/get-message-list-by-user-id');

  }
}
